File: Tatsuki_Yabe/Q8/Typescript/hashmap.ts
function missingNumber(nums: number[]): number {
  var numsMap = new Map<number, boolean>();
  var output = 0;
  nums.forEach((num) => {
    numsMap.set(num, true);
  });

  for (var i = 1; i <= nums.length; i++) {
    if (!numsMap.get(i)) {
      output = i;
    }
  }

  return output;
}

File: Tatsuki_Yabe/Q8/Swift/hashmap.swift
class Solution {
    func missingNumber(_ nums: [Int]) -> Int {
        var dictionary = [Int: Bool]()
        var output = 0
        nums.forEach { num in 
            dictionary[num] = true       
        }

        for i in 1 ... nums.count {
            if !(dictionary[i] ?? false) {
                output = i
            }
        }

        return output
    }
}